File: OpenSim Tools/ProcessingTrialDataFromVicon/rezero_filter.py
import numpy as np
import re

def rezero_filter(original_fy: np.ndarray):
	'''
	Resets all values which were originally zero to zero

	'''

	filter_plate = np.zeros(np.shape(original_fy))

	for i in range(len(original_fy[0,:])):
		# Binary test for values greater than 20
		force_zero = (original_fy[:,i] > 20) * 1 # Convert to 1 or 0 rather than True or False

		# We do not want to accept values which are over 20 but considered 'nosie'.
		# Must be over 20 for more than 10 frames in a row. Therefore, need to search for
		# [1, 1, 1, 1, 1, 1, 1, 1, 1, 1] and get all the indices that meet this condition

		# Convert to string to test this condition
		force_str = ''.join(list(map(str, force_zero)))

		# Find all occurrences where the pattern occurs
		true_inds = [m.start() for m in re.finditer('(?=1111111111)', force_str)]

		# true_inds will not include the ends (e.g., 11...11100000) - will not include the final 3 1's
		extra_inds = [i + 10 for i in true_inds[0:-1]] # So make another array with 10 added on to all but the last value
		
		# Return the 'filtered' rezeroing array
		filter_plate[true_inds,i] = 1
		filter_plate[extra_inds,i] = 1

	return filter_plate

File: ProcessingTrialDataFromVicon_MATLAB/reZeroFilter.m
function [filterPlate] = reZeroFilter(originalFys)
%reZeroForceData: Resets all values that were originally zero to zero.

for i = 1:length(originalFys(1,:))
    positiveFy = originalFys(:,i) > 20;
    
    % Finds index of 1's, only if there are at least 9 more 1's ahead of
    % it?
    trueInd = strfind(positiveFy',[1 1 1 1 1 1 1 1 1 1]);
    
    % But then adds 10 on to this - cancelling out what was just done?
    extraInd = trueInd(1:end-1) + 10;
    
    filterPlate(:,i) = zeros(length(originalFys(:,i)),1);
    filterPlate(trueInd,i) = 1;
    filterPlate(extraInd,i) = 1;
end